package.json

```diff
@@ -20,7 +21,8 @@
     "lint": "yarn run lint:eslint && yarn run lint:stylelint && yarn run lint:prettier",
     "lint:eslint": "eslint . --max-warnings 0 --cache --cache-location '.loom/cache/eslint'",
     "lint:stylelint": "yarn workspace '@shopify/stylelint-polaris' run gen-polaris-vars && stylelint '**/*.scss'",
-    "lint:prettier": "prettier '**/*.{md,json,yaml,yml}' --check"
+    "lint:prettier": "prettier '**/*.{md,json,yaml,yml}' --check",
+    "type-check": "tsc --build"
```

Comment:
**question:** Why do we run a build for the `type-check`? I was imagining this would be `tsc --noEmit`.

Reply:
It is a confusing flag name to use, but that is what the TypeScript docs suggest on their [project references page](https://www.typescriptlang.org/docs/handbook/project-references.html#build-mode-for-typescript) 🤷

Reply:
> Build mode (see below) will automatically build the referenced project if needed

Reply:
We want it to produce d.ts files when running type-check. This is because incremental builds - "hey you changed a file in the polaris-react project, typecheck the polaris-react project, and anything that depends polaris-react)" relies upon the d.ts files being present. We can ensure that `.d.ts`  are created, but no js files are created by using `"noEmit": false, "emitDeclarationOnly": true,` in the tsconfig files.

Reply:
Correct for files not using tsc to build - `"noEmit": false, "emitDeclarationOnly": true, "outDir": "some-dist"`

Files build with tsc would use `noEmit: false, emitDeclarationOnly:false`. (setting noEmit:true would stop all output. double negatives are confusing.

Reply:
Awesome, thanks for the breakdown @BPScott!!

@samrose3 Feel free to resolve this conversation